src/main/java/it/pagopa/interop/probing/eservice/registry/reader/dto/EserviceDTO.java

```diff
@@ -0,0 +1,90 @@
+/**************************************************************************
+*
+* Copyright 2023 (C) DXC
+*
+* Created on  : Feb 20, 2023
+* Author      : dxc technology
+* Project Name: interop-be-probing 
+* Package     : it.pagopa.interop.probing.dto
+* File Name   : EserviceDTO.java
+*
+*-----------------------------------------------------------------------------
+* Revision History (Release )
+*-----------------------------------------------------------------------------
+* VERSION     DESCRIPTION OF CHANGE
+*-----------------------------------------------------------------------------
+** --/1.0  |  Initial Create.
+**---------|------------------------------------------------------------------
+***************************************************************************/
+package it.pagopa.interop.probing.eservice.registry.reader.dto;
+
+import javax.validation.constraints.NotBlank;
+import javax.validation.constraints.NotEmpty;
+import javax.validation.constraints.Pattern;
+
+import it.pagopa.interop.probing.eservice.registry.reader.annotations.ValidateEnum;
+import it.pagopa.interop.probing.eservice.registry.reader.util.EServiceState;
+import it.pagopa.interop.probing.eservice.registry.reader.util.EserviceType;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+
+/**
+ * Gets the producer name.
+ *
+ * @return the producer name
+ */
+@Getter
+
+/**
+ * Sets the producer name.
+ *
+ * @param producerName the new producer name
+ */
+@Setter
+
+/**
+ * Instantiates a new eservice DTO.
+ */
+@NoArgsConstructor
+public class EserviceDTO {
+
+		/** The name. */
+		@NotBlank(message = "must not be blank")
+		private String name;
+		
+		/** The eservice id. */
+		@NotBlank(message = "must not be blank")
+		@Pattern(regexp="^[0-9a-f]{8}-[0-9a-f]{4}-[0-5][0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}$", message="must respect the UUID regex")
+		private String eserviceId;
+		
+		/** The version id. */
+		@Pattern(regexp="^[0-9a-f]{8}-[0-9a-f]{4}-[0-5][0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}$", message="must respect the UUID regex")
+		@NotBlank(message = "versionId must not be blank")
+		private String versionId;
+		
+		/** The type. */
+		@NotBlank(message = "must not be blank")
+		@ValidateEnum(
+			     enumClass = EserviceType.class,
+			     message = "value must be present in the EserviceType enum"
+			 )
+		private String type;
```

Comment:
We decided to use the String type so that we could validate the value present in the S3Object. If we change it with the enum, then a different exception from the validation one will be thrown ,that is the one of the failed mapping.

Reply:
OK

Reply:
Yes, we would have a deserialization exception because the deserializer would fail to map.So we use the type string and then we validate the value to be the one expected. By doing that we have a validation exception in case of a bad value , same behaviour as the other fields, with a custom message that we know.

Reply:
you mean the json library is not able to deserialize enums?

Reply:
i mean that if we receive a value that is not part of the enum values, of course the json library will give an exception , because it cannot be mapped into the enum.This way we have a far more explicit error,because the error is given during validation.

Reply:
ok if I got your point right, the issue here is just the validation library and the moment where the validation happens.
We can proceed for now and not stop because of this, but let's add it to out technical debt backlog, because type safety is preferable (and for example in case of enums the validation is free, without the need of additional libraries)